orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -0,0 +1,166 @@
+    this.deploymentMonitorServiceProvider = deploymentMonitorServiceProvider;
+    this.registry = registry;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.MINUTES.toMillis(1);
+  }
+
+  @Override
+  public long getTimeout() {
+    // TODO(mvulfson): Use DeploymentMonitorDefinition
+    return TimeUnit.MINUTES.toMillis(30);
+  }
+
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    TaskResult result;
+    MonitoredDeployStageData context = stage.mapTo(MonitoredDeployStageData.class);
+
+    try {
+      this.stage = stage;
+      this.monitorDefinition =
+          deploymentMonitorServiceProvider.getDefinitionById(
+              context.getDeploymentMonitor().getId());
+
+      result = executeInternal();
+    } catch (RetrofitError e) {
+      registry
+          .counter("deploymentMonitor.errors", "monitorId", monitorDefinition.getId())
+          .increment();
+
+      int currentRetryCount = context.getDeployMonitorHttpRetryCount();
+
+      if (currentRetryCount >= MAX_RETRY_COUNT) {
+        if (monitorDefinition.isFailOnError()) {
+          registry
+              .counter("deploymentMonitor.fatalErrors", "monitorId", monitorDefinition.getId())
+              .increment();
+
+          log.error(
+              "Failed to get valid response for {} from deployment monitor {}, aborting because the monitor is marked with failOnError",
+              getClass().getSimpleName(),
+              monitorDefinition);
+
+          result =
+              TaskResult.builder(ExecutionStatus.TERMINAL)
+                  // TODO(mvulfson)
+                  // .context()
+                  .build();
+        } else {
+          log.warn(
+              "Failed to get valid response for {} from deployment monitor {}, ignoring failure",
+              getClass().getSimpleName(),
+              monitorDefinition);
+
+          result =
+              TaskResult.builder(ExecutionStatus.SUCCEEDED)
+                  // TODO(mvulfson)
+                  // .context()
+                  .build();
+        }
+      } else {
+        // Retry
+        result =
+            TaskResult.builder(ExecutionStatus.RUNNING)
+                .context("deployMonitorHttpRetryCount", ++currentRetryCount)
+                .build();
+      }
+    } catch (Exception e) {
+      log.error("Exception while executing {}, aborting deployment", getClass().getSimpleName(), e);
+
+      // TODO(mvulfson): I don't love this
+      throw e;
+    }
+
+    return result;
+  }
+
+  public TaskResult executeInternal() {
+    throw new UnsupportedOperationException("Must implement executeInternal method");
+  }
+
+  void sanitizeAndLogResponse(EvaluateHealthResponse response) {
+    if ((response.getNextStep() == null) || (response.getNextStep().getDirective() == null)) {
+      log.error(
+          "Deployment monitor {}: returned invalid nextStep.directive, assuming ABORT",
```

Comment:
might as well log the `response.getNextStep()`, could be helpful

Reply:
The problem is that by this point I don't have the raw JSON value/string since Jackson already failed to deserialize it.... BUT... currently, these errors aren't logged nicely in the exception handler for deserialization - adding that now, thanks!